kolibri/core/webpack/hooks.py

```diff
@@ -0,0 +1,290 @@
+        """
+        for f in self.stats_file_content["files"]:
+            filename = f['name']
+            if any(regex.match(filename) for regex in settings.IGNORE_PATTERNS):
+                continue
+            relpath = '{0}/{1}'.format(self.unique_slug, filename)
+            f['url'] = staticfiles_storage.url(relpath)
+            yield f
+
+    @hooks.registered_method
+    def bundle_filtered(self, extension=None):
+        """
+        TODO: Why is this helper function necessary!?
+
+        :returns: a possibly filtered list of data from self.bundle
+        """
+        bundle = self.bundle
+        if extension:
+            bundle = (chunk for chunk in bundle if chunk['name'].endswith('.{0}'.format(extension)))
+        return bundle
+
+    @property
+    @hooks.registered_method
+    def webpack_bundle_data(self):
+        """
+        This is the main interface to the NPM Webpack building util. It is
+        used by the webpack_json management command. Inheritors may wish to
+        customize this.
+
+        :returns: A dict with information expected by webpack parsing process.
+        """
+        return {
+            "name": self.unique_slug,
+            "src_file": self.src_file,
+            "static_dir": self.static_dir,
+            "stats_file": self.stats_file,
+            "events": {},
+            "once": {},
+        }
+
+    @property
+    def build_path(self):
+        """
+        An auto-generated path to where the build-time files are stored,
+        containing information about the built bundles.
+        """
+        return os.path.join(
+            os.path.abspath(os.path.dirname(__name__)),
+            self._module_file_path,
+            "build"
+        )
+
+    @property
+    def stats_file(self):
+        """
+        TODO: Do we want to rely on a generated stats file? It will have to be
+        read for every bundle, every time stuff is loaded.
+
+        An auto-generated path to where the build-time files are stored,
+        containing information about the built bundles.
+        """
+        return os.path.join(
+            self.build_path,
+            "{plugin}_stats.json".format(plugin=self.unique_slug)
+        )
+
+    @property
+    def _module_file_path(self):
+        """
+        Returns the path of the class inheriting this classmethod.
+        """
+        return os.path.join(*self.__module__.split(".")[:-1])
+
+    def render_to_html(self, extension=None):
+        """
+        This function tags a bundle of file chunks and generates the appropriate
+        script tags for them, be they JS or CSS files.
+
+        :param bundle_data: The data returned from
+        :return: HTML of script tags for insertion into a page.
+        """
+        tags = []
+        js_tag = '<script type="text/javascript" src="{url}"></script>'
+        css_tag = '<link type="text/css" href="{url}" rel="stylesheet"/>'
+        for chunk in self.bundle_filtered(extension=extension):
```

Comment:
It could just be removed following the YAGNI ideal... and then re-added once a need appears. I was a bit confused by not knowing the exact intention, but I think it was also a good idea to have as much illustration of how things *could* essentially be done was where we are now.

I think that from here on, you are probably the main glue between the hooks API and servicing the frontend stuff.